backend/substrapp/compute_tasks/outputs.py

```diff
@@ -54,64 +55,47 @@ def _save_performance(self, output: context.OutputResource):
             }
             client.register_performance(performance_obj)
 
-    def _save_models(self, models: list[context.OutputResource]):
-        logger.info("saving models")
+    @retry()
+    def _save_model_to_local_storage(self, models: list[context.OutputResource]) -> bytes:
+        url = settings.DEFAULT_DOMAIN + reverse("api:model-list")
+        files = {}
+        data = {}
 
-        new_models = []
         for model in models:
-            new_model = self._save_model_to_local_storage(model)
-            new_models.append(new_model)
-
-        try:
-            with get_orchestrator_client(self._ctx.channel_name) as client:
-                registered_models = client.register_models({"models": new_models})
-        except Exception as exc:
-            for model in new_models:
-                self._delete_model(model["key"])
-            raise exc
-
-        for registered_model in registered_models:
-            registered_model["channel"] = self._ctx.channel_name
-            serializer = ModelRepSerializer(data=registered_model)
-            try:
-                serializer.save_if_not_exists()
-            except AlreadyExistsError:
-                pass
-
-        for model, local in zip(models, new_models):
             path = os.path.join(self._ctx.directories.task_dir, model.rel_path)
-            add_model_from_path(path, local["key"])
+            checksum = get_hash(path, self._ctx.task.key)
 
-    def _save_model_to_local_storage(self, model: context.OutputResource) -> dict:
-        path = os.path.join(self._ctx.directories.task_dir, model.rel_path)
-        from substrapp.models import Model
+            files[checksum + "-" + "file"] = open(path, "rb")
```

Comment:
How do you manage the closing of this file handler ?

Reply:
Good question ! 
I have to check if it's handle directly in the post requests (there is something like this if I remember well)